gammapy/spectrum/dataset.py

```diff
@@ -398,6 +409,66 @@ def create(cls, e_reco, e_true=None, reference_time="2000-01-01"):
         )
 
 
+    def stack(self, other):
+        """Stack this dataset with another one.
+
+        Safe mask is applied to compute the stacked counts vector.
+        Counts outside each dataset safe mask are lost.
+
+        Stacking is performed in-place.
+
+        Parameters
+        ----------
+        other : `~gammapy.spectrum.SpectrumDataset`
+            the dataset to stack to the current one
+       """
+
+        if not isinstance(other, SpectrumDataset):
+            raise TypeError("Incompatible types for SpectrumDataset stacking")
+
+        if self.counts is not None:
```

Comment:
I guess we should also add the check `and other.counts is not None:`?

Reply:
I wondered about this. we can indeed put `counts` to None if `other.counts is None`, but doing it silently seemed a problem to me. So I decided to let `stack` raise an error in this case. Another option would to add a `WARNING` in a log. 
Any preference?

Reply:
Thinking about it again, raising an error is probably fine. So just keep as is...